### PR TITLE
Add List RPC to UserRegistry

### PR DESCRIPTION
**Pull request by kschiffer**
#### Summary
References #1648 

This PR adds the `List` rpc to the UserRegistry.

#### Changes
- Add `List` rpc to api definition
- Add `ListUsers` handler to user registry in IS
- Regenerate protos

#### Notes for Reviewers
First time I'm working on the api / handlers, so watch out for obvious mistakes.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [x] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/ttnpb/field_mask_validation.go [hunk 1]
```diff
@@ -620,6 +620,7 @@ var AllowedFieldMaskPathsForRPC = map[string][]string{
 	// Users:
 	"/ttn.lorawan.v3.UserRegistry/Get":                 omitFields(UserFieldPathsNested, "password", "temporary_password"),
 	"/ttn.lorawan.v3.UserRegistry/Update":              omitFields(UserFieldPathsNested, "password", "password_updated_at"),
+	"/ttn.lorawan.v3.UserRegistry/List":                omitFields(UserFieldPathsNested, "password", "temporary_password"),
```
Comment (johanstokking) [inline pkg/ttnpb/field_mask_validation.go]: Use ascending sort

File: pkg/identityserver/user_registry.go [hunk 2]
```diff
@@ -258,6 +259,37 @@ func (is *IdentityServer) getUser(ctx context.Context, req *ttnpb.GetUserRequest
 	return usr, nil
 }
 
+func (is *IdentityServer) listUsers(ctx context.Context, req *ttnpb.ListUsersRequest) (users *ttnpb.Users, err error) {
+	req.FieldMask.Paths = cleanFieldMaskPaths(ttnpb.UserFieldPathsNested, req.FieldMask.Paths, getPaths, nil)
+	authInfo, err := is.authInfo(ctx)
```
Comment (htdvisser) [inline pkg/identityserver/user_registry.go]: We also have `is.IsAdmin` (in `entity_access.go`). Maybe nice to add a `is.RequireAdmin` (that returns an err) next to that.

File: pkg/identityserver/user_registry.go [hunk 3]
```diff
@@ -258,6 +259,37 @@ func (is *IdentityServer) getUser(ctx context.Context, req *ttnpb.GetUserRequest
...
+	}
+	var total uint64
+	paginateCtx := store.WithPagination(ctx, req.Limit, req.Page, &total)
+	defer func() {
+		if err == nil {
+			setTotalHeader(ctx, total)
+		}
+	}()
+	users = &ttnpb.Users{}
+	userIDs := make([]*ttnpb.UserIdentifiers, 0)
```
Comment (htdvisser) [inline pkg/identityserver/user_registry.go]: Can remove this

File: pkg/identityserver/user_registry.go [hunk 4]
```diff
@@ -258,6 +259,37 @@ func (is *IdentityServer) getUser(ctx context.Context, req *ttnpb.GetUserRequest
...
+	err = is.withDatabase(ctx, func(db *gorm.DB) error {
+		users.Users, err = store.GetUserStore(db).FindUsers(paginateCtx, userIDs, &req.FieldMask)
```
Comment (htdvisser) [inline pkg/identityserver/user_registry.go]: Can pass `nil` instead of `userIDs`

File: pkg/identityserver/user_registry.go [hunk 5]
```diff
@@ -64,6 +64,7 @@ var (
 	errPasswordStrengthUppercase = errors.DefineInvalidArgument("password_strength_uppercase", "need at least `{n}` uppercase letter(s)")
 	errPasswordStrengthDigits    = errors.DefineInvalidArgument("password_strength_digits", "need at least `{n}` digit(s)")
 	errPasswordStrengthSpecial   = errors.DefineInvalidArgument("password_strength_special", "need at least `{n}` special character(s)")
+	errPermissionDenied          = errors.DefinePermissionDenied("permission_denied", "unauthorized request to restricted resource")
```
Comment (htdvisser) [inline pkg/identityserver/user_registry.go]: This error is too generic to be in `user_registry.go`

File: pkg/ttnpb/field_mask_validation.go (hunk 1 above)
Comment (htdvisser) [reply pkg/ttnpb/field_mask_validation.go]: We typically order Create - Get - List - Update - Delete - Search (CRUDS), but I'm not sure if we do that consistently.

File: pkg/identityserver/user_registry.go (hunk 5 above)
Comment (htdvisser) [reply pkg/identityserver/user_registry.go]: Could go with the `is.RequireAdmin` I suggested in my comment on `listUsers`. After that, make sure to run `mage go:messages`.

File: pkg/identityserver/user_registry.go [hunk 6]
```diff
@@ -258,6 +258,33 @@ func (is *IdentityServer) getUser(ctx context.Context, req *ttnpb.GetUserRequest
 	return usr, nil
 }
 
+func (is *IdentityServer) listUsers(ctx context.Context, req *ttnpb.ListUsersRequest) (users *ttnpb.Users, err error) {
+	req.FieldMask.Paths = cleanFieldMaskPaths(ttnpb.UserFieldPathsNested, req.FieldMask.Paths, getPaths, nil)
+	err = is.RequireAdmin(ctx)
```
Comment (htdvisser) [inline pkg/identityserver/user_registry.go]: We would usually write this as

```go
if err = is.RequireAdmin(ctx); err != nil {
	return nil, err
}
```

**Pull request merged**